lib/ts/fetch.ts

```diff
@@ -254,6 +254,21 @@ export default class AuthHttpRequest {
             logDebugMessage("doRequest: Returning without interception");
             return await httpCall(config);
         }
+
+        const origHeaders = new Headers(
+            config !== undefined && config.headers !== undefined ? config.headers : url.headers
+        );
+
+        if (origHeaders.has("Authorization")) {
+            const accessToken = await getTokenForHeaderAuth("access");
+            if (accessToken !== undefined && origHeaders.get("Authorization") === `Bearer ${accessToken}`) {
+                logDebugMessage(
+                    "doRequest: Removing Authorization from user provided headers because it contains our access token"
+                );
+                origHeaders.delete("Authorization");
+            }
+        }
```

Comment:
should add a clear comment explaining why this is being done. Describe the scenario in detail.